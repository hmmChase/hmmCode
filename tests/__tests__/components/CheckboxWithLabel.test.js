import { render, prettyDOM, fireEvent } from '@testing-library/react';
import CheckboxWithLabel from '../../components/CheckboxWithLabel';

describe('CheckboxWithLabel', () => {
  it('matches snapshot', () => {
    const mockProps = {};

    const utils = render(<CheckboxWithLabel {...mockProps} />);

    const container = utils.container.firstChild;

    expect(container).toMatchSnapshot();
  });

  it('changes text after click', () => {
    const mockProps = {};
    const onChange = jest.fn();

    const utils = render(
      <CheckboxWithLabel {...mockProps} onChange={onChange} />
    );

    const input1 = utils.queryByLabelText('false');

    expect(input1).toBeTruthy();
    expect(input1).toBeInTheDocument();
    expect(input1).toHaveAttribute('type', 'checkbox');
    expect(input1).not.toBeChecked();

    const label1 = utils.getByText('false');

    fireEvent.click(label1);

    const input2 = utils.queryByLabelText('true');

    expect(input2).toBeChecked();
  });
});
