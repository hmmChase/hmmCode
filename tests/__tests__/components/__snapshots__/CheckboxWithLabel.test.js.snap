// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxWithLabel matches snapshot 1`] = `
<label>
  <input
    type="checkbox"
  />
  false
</label>
`;
